### PR TITLE
Add Spark 3.4.4 Shim 

### LINKED ISSUE #11662: [FEA] Support Apache Spark 3.4.4
**Is your feature request related to a problem? Please describe.**
Apache Spark 3.4.4 vote just passed and they will be releasing it soon.  Determine if we want to support it.

### PR DESCRIPTION
**Pull request by gerashegalov**
- Fix shimplify to execute `mvn generate-sources -Dshimplify=true -Dshimplify.move=true -Dshimplify.overwrite=true -Dshimplify.add.shim=344 -Dshimplify.add.base=343 -s jenkins/settings.xml`

Closes #11662

### CODE REVIEW

**Comment by gerashegalov** [comment]
400 failure is a known issue 
344 is waiting for spark-rapids-private

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

File: aggregator/pom.xml [hunk 1]
```diff
@@ -71,6 +71,28 @@
             <version>${spark-rapids-private.version}</version>
             <classifier>${spark.version.classifier}</classifier>
         </dependency>
+        <!--
+            placeholders for maximum number of delta-lake shims per Spark shim:
+            - if we need only N use rapids.delta.artifactIdN+1 = rapids.delta.artifactIdN
+        -->
```
Comment (jlowe) [inline aggregator/pom.xml:77]: Why is this refactoring part of an already large PR to add a single shim?  This belongs in a separate PR.

Comment (gerashegalov) [reply aggregator/pom.xml:77]: Because it is the use case that drives it. However, I see your point for making it separate

**Review by razajafri — DISMISSED** [review]
Besides Jason's comment, LGTM

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]
